internal/tasks/resourcesync.go

```diff
@@ -60,36 +62,87 @@ func (r *ResourceSync) Poll() {
 
 	for i := range resourcesyncs {
 		rs := &resourcesyncs[i]
-		defer r.updateResourceSyncStatus(rs)
-		reponame := rs.Spec.Data.Repository
-		repo, err := r.store.Repository().GetInternal(ctx, rs.OrgID, *reponame)
-		if err != nil {
-			// Failed to fetch Repository resource
-			addRepoNotFoundCondition(rs, err)
-			break
-		}
-		addRepoNotFoundCondition(rs, nil)
-		resources, err := r.parseAndValidateResources(ctx, rs, repo)
-		if err != nil {
-			log.Errorf("resourcesync/%s: parsing failed. error: %s", rs.Name, err.Error())
-			continue
-		}
+		_ = r.run(ctx, log, rs)
+	}
+}
+
+func (r *ResourceSync) run(ctx context.Context, log logrus.FieldLogger, rs *model.ResourceSync) error {
+	defer r.updateResourceSyncStatus(rs)
+	reponame := rs.Spec.Data.Repository
+	repo, err := r.store.Repository().GetInternal(ctx, rs.OrgID, *reponame)
+	if err != nil {
+		// Failed to fetch Repository resource
+		addRepoNotFoundCondition(rs, err)
+		return err
+	}
+	addRepoNotFoundCondition(rs, nil)
+	resources, err := r.parseAndValidateResources(ctx, rs, repo)
+	if err != nil {
+		log.Errorf("resourcesync/%s: parsing failed. error: %s", rs.Name, err.Error())
+		return err
+	}
+
+	owner := fmt.Sprintf("%s/%s", model.ResourceSyncKind, rs.Name)
+	fleets, err := r.parseFleets(resources, rs.OrgID, owner)
+	if err != nil {
+		err := fmt.Errorf("resourcesync/%s: error: %w", rs.Name, err)
+		log.Errorf("%e", err)
+		addResourceParsedCondition(rs, err)
+		return err
+	}
+	addResourceParsedCondition(rs, nil)
+
+	fleetsOwned, err := r.store.Fleet().ListOwnedBy(ctx, rs.OrgID, owner)
+	if err != nil {
+		err := fmt.Errorf("resourcesync/%s: failed to list owned fleets. error: %w", rs.Name, err)
+		log.Errorf("%e", err)
+		return err
+	}
+
+	fleetsToRemove := r.fleetsDelta(fleetsOwned.Items, fleets)
 
-		fleets, err := r.parseFleets(resources, rs.OrgID)
+	r.log.Infof("resourcesync/%s: applying #%d fleets ", rs.Name, len(fleets))
+	err = r.store.Fleet().CreateOrUpdateMultiple(ctx, rs.OrgID, r.taskManager.FleetTemplateRolloutCallback, fleets...)
+	if err == gorm.ErrInvalidData {
+		err = fmt.Errorf("one or more fleets are managed by a differen resource. %w", err)
+	}
+	if len(fleetsToRemove) > 0 {
+		r.log.Infof("resourcesync/%s: found #%d fleets to remove. removing\n", rs.Name, len(fleets))
+		err := r.store.Fleet().DeleteMultiple(ctx, rs.OrgID, fleetsToRemove...)
```

Comment:
Do we want to delete the fleet or just remove the owner?  Maybe only delete the fleet if it has no matching devices?

Reply:
as we discussed. if the resourcesync is deleted - we remove the owner from all owned resources (fleets). If a definition was removed, we expect the fleet to be removed as well.

But we can revisit this